lib/AudioPlayback/AudioPlayback.js

```diff
@@ -41,326 +35,351 @@ var
+					{name: 'btnJumpForward', kind: IconButton, publish: true, icon: 'skipforward', small: false, command: 'jumpForward', classes: 'moon-audio-icon-button left', ontap: 'commandHandler', onholdpulse: 'holdForwardHandler', onrelease: 'releaseHandler'}
+				]},
+				{name: 'moreControls', classes: 'moon-audio-more-buttons'}
+			]},
+			{rtl: false, classes: 'moon-audio-playback-bottom', components: [
+				{kind: Slider, enableJumpIncrement: false, spotlight: false, name: 'slider', classes: 'moon-audio-slider', 
+				knobClasses: 'moon-audio-slider-knob', bgBarClasses: 'moon-audio-slider-bg-bar', 
+				barClasses: 'moon-audio-slider-bar-bar', tapAreaClasses: 'moon-audio-slider-taparea',
+				rtl: false, noPopup: true, lockBar: true, onChanging: 'sliderChanging', onAnimateFinish: 'sliderChanging',
+				onmove: 'preview', onenter: 'enterTapArea', onleave: 'leaveTapArea'},
+				{name: 'timePlayed', classes: 'moon-audio-play-time left', content: '0:00'},
+				{name: 'timeRemaining', classes: 'moon-audio-play-time right', content: '0:00'}
+			]}
+		]}
+	],
 
 	/**
 	* @private
 	*/
-	open: false,
+	observers: [
+		{method: 'modelChanged', path: ['model']}
+	],
 
 	/**
 	* @private
 	*/
-	controlsOpen: false,
+	bindings: [
+		{from: 'showJumpControls', to: '$.btnJumpBackward.showing'},
+		{from: 'showJumpControls', to: '$.btnJumpForward.showing'},
+		{from: 'showFFRewindControls', to: '$.btnRewind.showing'},
+		{from: 'showFFRewindControls', to: '$.btnFastForward.showing'},
+		{from: 'disableSlider', to: '$.slider.disabled'},
+		{from: 'disablePlaybackControls', to: '$.btnJumpBackward.disabled'},
+		{from: 'disablePlaybackControls', to: '$.btnJumpForward.disabled'},
+		{from: 'disablePlaybackControls', to: '$.btnRewind.disabled'},
+		{from: 'disablePlaybackControls', to: '$.btnFastForward.disabled'},
+		{from: 'disablePlaybackControls', to: '$.btnPlay.disabled'},
+		{from: 'jumpSec', to: '$.media.jumpSec'}
+	],
 
 	/**
 	* @private
 	*/
-	handlers: {
-		onAudioEnd: 'audioEnd'
+	create: function () {
+		this.inherited(arguments);
+		this.$.controlDrawer.createComponents(this.mediaComponents, {owner: this});
+		if (this.moreComponents) {
+			var owner = this.hasOwnProperty('moreComponents') ? this.getInstanceOwner() : this;
+			this.$.moreControls.createComponents(this.moreComponents, {owner: owner, ontap: 'commandHandler'});
+		}
+		this.playbackRateHashChanged();
 	},
 
 	/**
+	* Facade playbackRateHash.
 	* @private
 	*/
-	audioTracks: [],
+	playbackRateHashChanged: function() {
+		this.$.media.playbackRateHash = this.playbackRateHash;
+	},
 
-	/**
-	* @private
-	*/
-	index: null,
+	/** preview mode */
 
 	/**
 	* @private
 	*/
-	controlDrawerComponents:[],
+	enterTapArea: function (sender, event) {
+		if (event.originator.name == 'tapArea') {
+			this.startPreview();
+			if (!this.$.slider.disabled && this.hideButtonsOnSlider) {
+				this.$.playbackControls.setShowing(false);
```

Comment:
We should use the newer `this.$.playbackControls.set('showing', false);` in these instead. There's several of these, I won't point out each of them.